src/Microsoft.ML.Transforms/CategoricalHashTransform.cs

```diff
@@ -135,100 +136,123 @@ public sealed class Arguments : TransformInputBase
-
-                // creating the Hash function
-                var hashArgs = new HashTransformer.Arguments
-                {
-                    HashBits = args.HashBits,
-                    Seed = args.Seed,
-                    Ordered = args.Ordered,
-                    InvertHash = args.InvertHash,
-                    Column = new HashTransformer.Column[args.Column.Length]
-                };
-                for (int i = 0; i < args.Column.Length; i++)
-                {
-                    var column = args.Column[i];
-                    if (!column.TrySanitize())
-                        throw h.ExceptUserArg(nameof(Column.Name));
-                    h.Assert(!string.IsNullOrWhiteSpace(column.Name));
-                    h.Assert(!string.IsNullOrWhiteSpace(column.Source));
-                    hashArgs.Column[i] = new HashTransformer.Column
-                    {
-                        HashBits = column.HashBits,
-                        Seed = column.Seed,
-                        Ordered = column.Ordered,
-                        Name = column.Name,
-                        Source = column.Source,
-                        InvertHash = column.InvertHash,
-                    };
-                }
+                var col = new CategoricalHashEstimator.ColumnInfo(
+                    column.Source ?? column.Name,
+                    column.Name,
+                    column.OutputKind ?? args.OutputKind,
+                    column.HashBits ?? args.HashBits,
+                    column.Seed ?? args.Seed,
+                    column.Ordered ?? args.Ordered,
+                    column.InvertHash ?? args.InvertHash);
+                columns.Add(col);
+            }
+            return new CategoricalHashEstimator(env, columns.ToArray()).Fit(input).Transform(input) as IDataTransform;
+        }
+
+        private readonly TransformerChain<ITransformer> _transformer;
 
-                return CreateTransformCore(
-                    args.OutputKind, args.Column,
-                    args.Column.Select(col => col.OutputKind).ToList(),
-                    HashTransformer.Create(h, hashArgs, input),
-                    h,
-                    args);
+        public CategoricalHashTransform(HashEstimator hash, IEstimator<ITransformer> keyToVector, IDataView input)
+        {
+            var chain = hash.Append(keyToVector);
+            _transformer = chain.Fit(input);
+        }
+
+        public ISchema GetOutputSchema(ISchema inputSchema) => _transformer.GetOutputSchema(inputSchema);
+
+        public IDataView Transform(IDataView input) => _transformer.Transform(input);
+
+        public void Save(ModelSaveContext ctx) => _transformer.Save(ctx);
+
+        public bool IsRowToRowMapper => _transformer.IsRowToRowMapper;
+
+        public IRowToRowMapper GetRowToRowMapper(ISchema inputSchema)
+        {
```

Comment:
>{ [](start = 8, length = 1)

`=>` as well? It should check

Reply:
what?

---
In reply to: [220399209](https://github.com/dotnet/machinelearning/pull/1033#discussion_r220399209) [](ancestors = 220399209)

Reply:
I mean the inner method will check for the schema anyway
so you can use `=>` here

---
In reply to: [221013063](https://github.com/dotnet/machinelearning/pull/1033#discussion_r221013063) [](ancestors = 221013063,220399209)